tests/unit/test_container.py

```diff
@@ -456,6 +475,74 @@ def test_repr_html_(self):
         )
 
 
+    def test_repr_html_hdf5_dataset(self):
```

Comment:
@stephprince I added a test.

Reply:
The current example sort of works, is there something missing?

Reply:
ah you're right it was working fine, I was just getting a test failure from the docval data type in the `ContainerWithData` initialization so I changed it to be the more general "array_data" type.

I also added tests for numpy and Zarr arrays. Let me know if those look ok to you.

Reply:
The current tests look godo to me, thanks.